rockchip: add Radxa E25 support

maurerr opened the pull request:
Radxa E25 is a network application carrier board for the Radxa CM3
Industrial (CM3I) SoM, which is based on the Rockchip RK3568 SoC.

It has the following features:

- MicroSD card socket, on board eMMC flash
- 2x 2.5GbE Realtek RTL8125B Ethernet transceiver
- 1x USB Type-C port (Power and Serial console)
- 1x USB 3.0 OTG port
- mini PCIe socket (USB or PCIe)
- ngff PCIe socket (USB or SATA)
- 1x User LED and 16x RGB LEDs
- 26-pin expansion header

Installation:
Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
internal eMMC using dd.

Signed-off-by: Marius Durbaca <mariusd84@gmail.com>

maurerr commented:
cc @1715173329 @PolynomialDivision - would you be so kind to review my PR? I Believe I've incorporated all the lessons learned from my previous CM3 IO PR review. Thanks !

1715173329 commented:
oops, i did the same work to immortalwrt about an hour ago :P

maurerr commented:
> oops, i did the same work to immortalwrt about an hour ago :P

I was actually about to sent a PR to immortalwrt for you to help me troubleshoot some eth not coming up last week but sorted it out by myself

here is the dmesg log https://github.com/openwrt/openwrt/pull/14740#issuecomment-1966744907

1715173329 commented:
> here is the dmesg log https://pastebin.com/F4huhNZN

oh thanks for the log
looks like radxa somehow wrote a mac into the ethernet and i think overwrite it using mmc id is not that necessary

maurerr commented on target/linux/rockchip/armv8/base-files/etc/board.d/02_network:
```diff
@@ -43,7 +44,8 @@ rockchip_setup_macs()
 	friendlyarm,nanopi-r4s|\
-	friendlyarm,nanopi-r5s)
+	friendlyarm,nanopi-r5s|\
+	radxa,e25)
```
so not needed according to the log

maurerr replied:
done

1715173329 commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -104,6 +104,16 @@ define Device/radxa_cm3-io
+  SOC := rk3568
+  DEVICE_DTS := rockchip/rk3568-radxa-e25
+  UBOOT_DEVICE_NAME := radxa-e25-rk3568
+  DEVICE_PACKAGES := kmod-r8169
```
it could be done in the same PR I think

maurerr replied:
done

maurerr committed:
kernel: kmod-ata-ahci-platform: enable support for RK35xx

enable support for RK35xx in kmod-ata-ahci-platform kernel module

Suggested-by: Tianling Shen <cnsztl@immortalwrt.org>
Signed-off-by: Marius Durbaca <mariusd84@gmail.com>

maurerr committed:
uboot-rockchip: add Radxa E25 board support

add Radxa E25 board support in uboot-rockchip

Signed-off-by: Marius Durbaca <mariusd84@gmail.com>
Reviewed-by: Tianling Shen <cnsztl@immortalwrt.org>

maurerr commented on target/linux/rockchip/patches-6.1/025-v6.3-arm64-dts-rockchip-Update-eMMC-SD.patch:
```diff
@@ -0,0 +1,52 @@
+ arch/arm64/boot/dts/rockchip/rk3568-radxa-e25.dts     | 3 +--
+ 2 files changed, 3 insertions(+), 4 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-vmarc-som.dtsi b/arch/arm64/boot/dts/rockchip/rk3399pro-vmarc-som.dtsi
```
done

maurerr committed:
rockchip: add Radxa E25 board support

Radxa E25 is a network application carrier board for the Radxa CM3
Industrial (CM3I) SoM, which is based on the Rockchip RK3568 SoC.

It has the following features:

- MicroSD card socket, on board eMMC flash
- 2x 2.5GbE Realtek RTL8125B Ethernet transceiver
- 1x USB Type-C port (Power and Serial console)
- 1x USB 3.0 OTG port
- mini PCIe socket (USB or PCIe)
- ngff PCIe socket (USB or SATA)
- 1x User LED and 16x RGB LEDs
- 26-pin expansion header

Installation:
Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
internal eMMC using dd.

Signed-off-by: Marius Durbaca <mariusd84@gmail.com>

maurerr commented:
> for the second commit, there should be a newline between your description and SoB:
> 
> > add Radxa E25 board support in uboot-rockchip
> > Signed-off-by: Marius Durbaca [mariusd84@gmail.com](mailto:mariusd84@gmail.com)
> 
> Otherwise: `Reviewed-by: Tianling Shen <cnsztl@immortalwrt.org>`
> 
> Thank you!

done. thank you too !

1715173329 commented:
Cc @hauke

1715173329 commented:
kindly ping @hauke @PolynomialDivision any chance to take a look?

PolynomialDivision commented:
LGTM. I will merge.

PolynomialDivision commented:
@maurerr Oh, just saw something. Please add the install section also in the commit message.

maurerr commented:
> @maurerr Oh, just saw something. Please add the install section also in the commit message.

done

ETE-Design commented:
@1715173329 Can you pls add this to the OpenWrt Firmware Selector? I only own Windows Machine so is not possible to compile for me :-(

maurerr commented:
>  Can you pls add this to the OpenWrt Firmware Selector? I only own Windows Machine so is not possible to compile for me :-(
![image](https://github.com/openwrt/openwrt/assets/7603227/c7ce41f9-8852-4195-9b9b-610fae559a24)

ETE-Design commented:
@maurerr Seems like only upgrade image is possible... No factory image...

ETE-Design commented:
@maurerr Can you help on how to install it on eMMC? Seems like Radxa forgot that part when your on a Windows Machine... I have looked here: https://wiki.radxa.com/Rock3/installusb-install-radxa-e25 but seems like they are missing the part about how to upload it when your on Windows.

ETE-Design commented:
@maurerr Seems like it takes forever to get it out of Snapshot :-( Is there a way to get LuCi installed / downloaded from the start so I don't have to fiddle around with SSH ans so?

maurerr commented:
> @maurerr Seems like it takes forever to get it out of Snapshot :-( Is there a way to get LuCi installed / downloaded from the start so I don't have to fiddle around with SSH ans so?

of course - just use the firmware selector as it has luci included https://firmware-selector.openwrt.org/?version=SNAPSHOT&target=rockchip%2Farmv8&id=radxa_e25
also upcoming 24.10 snapshots have luci installed https://downloads.openwrt.org/releases/24.10-SNAPSHOT/targets/rockchip/armv8/
also next time please ask on https://forum.openwrt.org/
